src/core/meshcop/dataset.cpp

```diff
@@ -194,6 +200,7 @@ bool Dataset::ContainsAllRequiredTlvsFor(Type aType) const
     static const Tlv::Type kDatasetTlvs[] = {
         Tlv::kActiveTimestamp,
         Tlv::kChannel,
+        // Tlv::kWakeupChannel,
```

Comment:
Remove this commented line?

Reply:
This raises a question about backward compatibility with the new "wakeup channel TLV" in the dataset.

Consider a new device, supporting the "wakeup" feature, that joins an existing Thread mesh. The existing devices are running earlier Thread version, and their Dataset doesn't include this TLV. What should the new device do? Should it fail to join? Use some default value? Attempt to update the dataset?

This might be something to discuss within the Thread Tech committee. It might also be worth filing a JIRA ticket to discuss and clarify this further.

Thoughts?

Reply:
Submitted [SPEC-1324](https://threadgroup.atlassian.net/browse/SPEC-1324)

Reply:
I let the commented line here on purpose to raise this discussion. I'll remove it for now (since it was not part of the prototype, which defaults to channel 11) . We can adopt the final behavior later on.